develop/application-devs/smart-contracts/smart-contracts.md

```diff
@@ -0,0 +1,165 @@
+
+The Polkadot SDK presently supports smart contracts out-of-the-box in several ways:
+
+- The EVM pallet offered by [Frontier](https://github.com/paritytech/frontier){target=\_blank}
+- The [contracts pallet](https://github.com/paritytech/polkadot-sdk/blob/master/substrate/frame/contracts/){target=\_blank} in the FRAME library for Wasm-based contracts
+
+### Frontier EVM Contracts
+
+[Frontier](https://github.com/paritytech/frontier) is the suite of tools that enables a Substrate chain to run Ethereum contracts (EVM) natively with the same API/RPC interface, Ethereum exposes on Substrate. Ethereum Addresses can also be mapped directly to and from Substrate's SS58 scheme from existing accounts.
+
+- [Pallet EVM](https://docs.rs/pallet-evm/latest/pallet_evm/){target=\_blank} - enables functionality of running EVM (Solidity) contracts
+- [Pallet Ethereum](https://docs.rs/pallet-ethereum/latest/pallet_ethereum/){target=\_blank} - Ethereum RPC implementation, block emulation, and Ethereum transaction validation
+
+### Contracts Pallet
+
+The contracts pallet (`pallet_contracts`) implements a WebAssembly based approach to smart contracts.
+
+1. **Wasm**. The contracts pallet uses WebAssembly as its compilation target. Any language that
+   compiles to Wasm can potentially be used to write smart contracts. Nevertheless, it is better to
+   have a dedicated domain-specific language, and for that reason Parity offers the [ink!](#ink)
+   language.
+
+2. **Deposit**. Contracts must hold a deposit (named _ContractDeposit_ ) suitably large enough in
+   order to justify their existence on-chain. The developer needs to deposit this into the new contract
+   on top of the _ExistentialDeposit_.
+
+3. **Caching**. Contracts are cached by default and therefore means they only need to be deployed
+   once and afterward be instantiated as many times as you want. This helps to keep the storage load
+   on the chain down to the minimum. On top of this, when a contract is no longer being used and the
+   _existential deposit_ is drained, the code will be erased from storage (known as reaping).
+
+### Ink
+
+[ink!](https://github.com/use-ink/ink){target=\_blank} is a domain specific language for writing smart contracts
+in Rust and compiles to Wasm code.
+
+Here is the list of current resources available to developers who want to get started writing smart
+contracts to deploy on parachains based on Substrate.
+
+- [ink!](https://use.ink/){target=\_blank} - Polkadot's "native" smart contract stack, based on WebAssembly
+- [OpenBrush](https://docs.openbrush.io/){target=\_blank}: an `ink!` library providing standard contracts based on [PSP](https://github.com/w3f/PSPs){target=\_blank} with useful contracts and macros for building
+- [ink!athon](https://inkathon.xyz/){target=\_blank}: Starter kit for full-stack dApps with ink! smart contracts and frontend
+
+ink! has laid much of the groundwork for a new smart contract stack that is based on a Wasm virtual
```

Comment:
Throughout this page, I've read both `WebAssembly` and `Wasm`, please stick to one standard